### PR TITLE
Support for TopN for streaming

### LINKED ISSUE #294: Document the support of TopN for streaming (mv)
Only rank() but not dense_rank().

### Related code PR

https://github.com/risingwavelabs/risingwave/pull/5116

### Which part(s) of the docs might be affected or should be updated? And how?

Select command or a new page about TopN.

## Test cases

statement ok
create table t(x int, y int);

statement ok
create materialized view mv as 
select x, y from (
    select *, ROW_NUMBER() OVER (PARTITION BY x ORDER BY y) as rank from t
)
where rank <= 3;

statement ok
insert into t values 
    (1, 1), (1, 2), (1, 3), (1, 4), (1, 5),
    (2, 1), (2, 2), (2, 3), (2, 4), (2, 5),
    (3, 1), (3, 2), (3, 3), (3, 4), (3, 5);

query II rowsort
select * from mv;
----
1  1
1  2
1  3
2  1
2  2
2  3
3  1
3  2
3  3

### Reference

https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/dev/table/sql/queries/topn/

### PR DESCRIPTION
**Pull request by bernscode**
Support for TopN for streaming.

## Info
- **Description**: 
Add support for Top-N for streaming.

- **Related code PR**: 
https://github.com/risingwavelabs/risingwave/pull/5116

- **Related doc issue**: 
Resolves https://github.com/risingwavelabs/risingwave-docs/issues/294

- **Notes**: 
[ Any additional information? ]

## Before merging
  - [ ] (For version-specific PR) I have selected the corresponding software version in **Milestone** and linked the related doc issue to this PR in **Development**.
  - [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`bernscode`, `CharlieSYH`, `emile-00`, & `hengm3467`). 
  - [ ] I have checked the doc site preview, and the updated parts look good. <details><summary>How?</summary>image</details>

## Published doc pages
  [ After merging this PR, edit the description to include the links to the updated doc pages here. For example, https://www.risingwave.dev/docs/latest/intro/. ]

### CODE REVIEW

**Review by xxchan — COMMENTED** [review]
[Flink](https://nightlies.apache.org/flink/flink-docs-release-1.15/docs/dev/table/sql/queries/topn/) puts this section under "Queries/Top-N" and [materialize](https://materialize.com/docs/sql/patterns/top-k/) puts this section under "SQL patterns/Top K by Group" and I think it's a good idea to put it on a dedicated page (and also include an example).

File: docs/sql/commands/sql-select.md [hunk 1]
```diff
@@ -88,4 +88,35 @@ WHERE taxi_id IN
     WHERE company_id = 'Yellow Taxi')
 GROUP BY taxi_id
 ORDER BY total_distance, total_duration;
-```
\ No newline at end of file
+```
+
+## Top-N
+Top-N queries provide the N largest or smallest values ordered by columns. Top-N queries are helpful to display only the N top-most or the N bottom-most records from the batch/streaming table on a condition.
```
Comment (xxchan) [inline docs/sql/commands/sql-select.md]: `from the batch/streaming table` is a little bit confusing. We don't have such concepts, but table & materialized view.

File: docs/sql/commands/sql-select.md [hunk 2]
```diff
@@ -88,4 +88,35 @@ WHERE taxi_id IN
...
+
+Syntax of the Top-N statement:
+```sql
+SELECT [column_list]
```
Comment (xxchan) [inline docs/sql/commands/sql-select.md]: Better to add a note below telling that `rank` cannot be included in `column_list`

File: docs/sql/commands/sql-select.md [hunk 3]
```diff
@@ -88,4 +88,35 @@ WHERE taxi_id IN
...
+        ORDER BY col1 [asc|desc][, col2 [asc|desc]...]) AS rank 
+    FROM table_name)
+WHERE rank <= N;
+```
+
+Parameter Specification:
+
+<ul>
+  <li>
+    ROW_NUMBER(): Assigns a unique, sequential number to each row, starting with one, according to the ordering of rows within the partition.
```
Comment (xxchan) [inline docs/sql/commands/sql-select.md]: RANK is also supported now. Do we document it together now?

BTW, currently status is:
- streaming (CREATE MATERIALIZED VIEW): ROW_NUMBER ✅ RANK ✅
- batch: ROW_NUMBER ✅ RANK ❌

File: docs/sql/commands/sql-select.md (hunk 1 above)
Comment (bernscode) [reply docs/sql/commands/sql-select.md]: @xxchan Good point! I've removed this part from the latest commit. Thanks!

File: docs/sql/commands/sql-select.md (hunk 2 above)
Comment (bernscode) [reply docs/sql/commands/sql-select.md]: @xxchan Great idea! I've added this note below the example on the latest commit. Thanks!

File: docs/sql/commands/sql-select.md (hunk 3 above)
Comment (bernscode) [reply docs/sql/commands/sql-select.md]: @xxchan Excellent! I've added this to the ROW_NUMBER() info on the new commit. Thanks!

**Comment by xxchan** [comment]
replaced by #317

**Pull request closed without merging**